### PR TITLE
CI, Mac os: Use the qt-install-action to install Qt

### PR DESCRIPTION
**Pull request by sithlord48**
- Fix the macOS ci by using qt install action to install qt

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -171,16 +171,34 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [macos-11, macos-12]
+       include:
+         - os: macos-11
+           qt-version: 5.15
+           qt-major: 5
```
Comment (shymega) [inline .github/workflows/builds.yml:177]: I don't like repeating the variables here. Can we use any of the string manipulation functions available in GH Actions to extract the first '5.' or '6.' in `qt-version`?

That way, we can just use one variable.

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -171,16 +171,34 @@ jobs:
...
 
     env:
       VERBOSE: 1
       TERM: xterm-256color
 
     steps:
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v3
+        with:
+          dir: /Users/runner
```
Comment (shymega) [inline .github/workflows/builds.yml:194]: I think I'd rather use a GH Actions context variable here.

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -190,10 +208,9 @@ jobs:
 
       - name: Setup the build
         run: |
-          . input-leap/macos_environment.sh
```
Comment (shymega) [inline .github/workflows/builds.yml:193]: @p12tic Are you happy with this?

**Review by shymega — CHANGES_REQUESTED** [review]
Just a few things.

Also, will this affect the pending PR of #1713?

File: .github/workflows/builds.yml (hunk 2 above)
Comment (sithlord48) [reply .github/workflows/builds.yml:194]: In this case there is no reason we don't use it again.
In the future i plan to unify the steps and use this action for all at that point i will use a variable

**Comment by sithlord48** [comment]
> Just a few things.
> 
> Also, will this affect the pending PR of #1713?

This has been cherry picked from #1713 to pull only the qt action since this fixes our Ci build due to os11 taking forever to checkout qt5 from homebrew

File: .github/workflows/builds.yml (hunk 1 above)
Comment (sithlord48) [reply .github/workflows/builds.yml:177]: I plan to improve such things when i unify all the build steps

Comment (shymega) [reply .github/workflows/builds.yml]: I think I'd be in favour of improving these things at the beginning.

File: .github/workflows/builds.yml (hunk 2 above)
Comment (shymega) [reply .github/workflows/builds.yml]: I think I'd be in favour of improving these things at the beginning.

Comment (sithlord48) [reply .github/workflows/builds.yml]: Un needed so remvoed

File: .github/workflows/builds.yml (hunk 3 above)
Comment (shymega) [reply .github/workflows/builds.yml:193]: Scratch this. I'm happy with it.

**Review by shymega — APPROVED** [review]
Now my requested changes have been actioned, I'm happy with this PR.

LGTM. Merge pending on @p12tic's review, but it would resolve the `macos-11` issues we have now.

**Comment by shymega** [comment]
@p12tic Just need your review now. Commit message seems alright to me, could do with some more description, but not enough of an issue to warrant me to review.

**Pull request merged**